Formula/pnpm.rb

```diff
@@ -20,11 +20,65 @@ class Pnpm < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "efcac832dc165a64bbb93b7018876cf7dc70d669ae3b1f2a68c46ea0bcddc572"
   end
 
-  depends_on "node"
+  depends_on "node" => :build
+
+  PNPM_BUILDTIME_VERSION = "v6.7".freeze
+
+  resource "pnpm-buildtime" do
+    url "https://get.pnpm.io/#{PNPM_BUILDTIME_VERSION}.js"
+    sha256 "56bffe269aab055bec3a07fa65c387f5a35729550dc7fc76711844a2de6e1b8b"
+  end
 
   def install
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
+    resource("pnpm-buildtime").stage do
+      (buildpath/"buildtime-bin").install "#{PNPM_BUILDTIME_VERSION}.js"
+      IO.write "pnpm", <<~EOS
+        #!/bin/sh
+        node #{buildpath/"buildtime-bin"/PNPM_BUILDTIME_VERSION}.js "$@"
+        exit "$?"
+      EOS
+      chmod 0755, "pnpm"
+      (buildpath/"buildtime-bin").install "pnpm"
+    end
+    ENV.prepend_path "PATH", buildpath/"buildtime-bin"
+    system "pnpm", "install"
+    system "pnpm", "run", "compile-only"
+    on_macos do
+      if Hardware::CPU.arm?
+        chdir "packages/pnpm" do
+          mkdir_p "../artifacts/macos-arm64"
+          system "node_modules/.bin/pkg", "./dist/pnpm.cjs",
+            "--out-path=../artifacts/macos-arm64", "--targets=node14-macos-arm64"
+        end
+        bin.install "packages/artifacts/macos-arm64/pnpm"
+      else
+        bin.install "packages/artifacts/macos-x64/pnpm"
+      end
+    end
+    on_linux do
+      bin.install "packages/artifacts/linux-x64/pnpm"
+    end
+  end
+
+  def caveats
+    <<~EOS
+      You should create npm's global directory if you want to install packages globally:
+
+        mkdir -p ~/npm-global/bin
+
+      Run the following pnpm command:
+
+        pnpm set prefix ~/npm-global
+
+      Add the following to #{shell_profile} or your desired shell
+      configuration file:
+
+        export PATH="$HOME/npm-global/bin:$PATH"
+
+      You can set prefix to any location, but leaving it unchanged from
+      #{HOMEBREW_PREFIX} will destroy any Homebrew-installed Node installations
+      upon upgrade/reinstall.
```

Comment:
Most of this should be in upstream documentation.

Reply:
Sure.

Reply:
as I noted above. setting that setting is not needed.